hoomd/compute.py

```diff
@@ -191,71 +86,148 @@ class thermo(_compute):
 
     Examples::
 
-        g = group.type(name='typeA', type='A')
-        compute.thermo(group=g)
+        f = filter.Type('A')
+        compute.ThermoQuantities(filter=f)
     """
 
-    def __init__(self, group):
-
-        # initialize base class
-        _compute.__init__(self);
-
-        suffix = '';
-        if group.name != 'all':
-            suffix = '_' + group.name;
+    def __init__(self, filter):
+        super().__init__(filter)
 
-        # warn user if an existing compute thermo already uses this group or name
-        for t in hoomd.context.current.thermos:
-            if t.group is group:
-                hoomd.context.current.device.cpp_msg.warning("compute.thermo already specified for this group");
-            elif t.group.name == group.name:
-                hoomd.context.current.device.cpp_msg.warning("compute.thermo already specified for a group with name " + str(group.name) + "\n");
+    def _attach(self):
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            thermo_cls = _hoomd.ComputeThermo
+        else:
+            thermo_cls = _hoomd.ComputeThermoGPU
+        group = self._simulation.state.get_group(self._filter)
+        self._cpp_obj = thermo_cls(self._simulation.state._cpp_sys_def, group, "")
+        super()._attach()
+
+    @log
+    def temperature(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getTemperature()
+        else:
+            return None
 
-        # create the c++ mirror class
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            self.cpp_compute = _hoomd.ComputeThermo(hoomd.context.current.system_definition, group.cpp_group, suffix);
+    @log
+    def pressure(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPressure()
         else:
-            self.cpp_compute = _hoomd.ComputeThermoGPU(hoomd.context.current.system_definition, group.cpp_group, suffix);
+            return None
 
-        hoomd.context.current.system.addCompute(self.cpp_compute, self.compute_name);
+    @log
+    def pressureXX(self):
```

Comment:
Define just one `pressure_tensor` method.

Reply:
Agreed.

Reply:
I'm not completely up to date on the newest abilities of the logger, but won't logging a list quantity be a problem?

Reply:
No, the new logger can handle arbitrary data quantities, up to and including large numpy arrays with the number of particles in the system or larger. A single 6-element tensor is trivially handled.